inc/classes/class-rocket-database-optimization.php

```diff
@@ -66,16 +78,16 @@ public static function init() {
 	 *
 	 * @since 2.11
 	 * @author Remy Perona
+	 *
+	 * @param array $options WP Rocket Database options.
 	 */
-	public function process_handler() {
+	private function process_handler( $options ) {
 		if ( method_exists( $this->process, 'cancel_process' ) ) {
 			$this->process->cancel_process();
 		}
 
-		foreach ( $this->options as $option ) {
-			if ( get_rocket_option( 'database_' . $option, false ) ) {
-				$this->process->push_to_queue( $option );
-			}
+		foreach ( $options as $option ) {
+			$this->process->push_to_queue( $option );
```

Comment:
Or could be: `array_map( array( $this->process, 'push_to_queue' ), $options );`.